File: Dart/lib/longestPalindrome.dart
class Solution {
  int longestPalindrome(String s) {
    var map = <String, int>{};
    var sum = 0;
    var res = 0;
    for (var i = 0; i < s.length; i++) {
      String c = s[i];
      map[c] = (map[c] ?? 0) + 1;
    }
    map.forEach((key, value) {
      sum += value & ~1;
      res |= value & 1;
    });
    return sum + res;
  }
}

File: Ruby/solution/longest_palindrome.rb
# @param {String} s
# @return {Integer}
def longest_palindrome(s)
  counts = s.each_char.with_object(Hash.new(0)) { |c, h| h[c] += 1 }
  counts.values.sum { |c| c.even? ? c : c - 1 } + (counts.values.any?(&:odd?) ? 1 : 0)
end
